Doc/library/argparse.rst

```diff
@@ -977,6 +977,11 @@ the various :class:`ArgumentParser` actions.  The two most common uses of it are
   ``action='store_const'`` or ``action='append_const'``.  These actions add the
   ``const`` value to one of the attributes of the object returned by
   :meth:`~ArgumentParser.parse_args`. See the action_ description for examples.
+  If ``const`` is not provided to :meth:`~ArgumentParser.add_argument`, it
+  will receive a default value of ``None``.
+
+   .. versionchanged:: 3.10
+      ``const=None`` by default.
```

Comment:
Should this be one level higher to clarify it doesn't just apply to store_const and append_const? The part being removed below made that more clear.

If so, then the versionchanged could be more specific about what was changed in 3.11. (BTW 3.11 is under dev now)

Reply:
I will take another pass at the documentation, because I'm seeing a few rough edges. However, to be clear, the default assignment of `None` to the const kwarg is *only* new behavior when the action type is append_const or store_const. All other action types **_always_** assigned the default value of `None` to const. This PR just makes the behavior consistent when the action type is `append_const` or `store_const`, so the change in behavior is limited in scope.

At the same time, I think that sticking the :versionchanged: note under the bullet point like I did initially just plain looks wrong, so I will bring the note back to its original position.

As far as "other rough edges":

- In the `action` section:
  - Grammatical change: exchanging parenthesis for a semicolon where there were two separate clauses in the `action='append_const'` section.
  - Add a brief note about the default assignment of `const=None` in the `action='store_const'` section.
- Add a paragraph break in the `name or flags` section. This makes it much easier for the reader to draw contrast between these two examples at a glance.